### DISCUSSION TITLE
Formula - Nested relations

### DISCUSSION THREAD

**Discussion by pbendersky** [Budibase/budibase#9302] [Help] [answered]
I'm trying to come up with a formula to calculate the total of an invoice.

My hierarchy of tables is as follows:

```
Invoice
   many InvoiceItem
InvoiceItem
   amount
   one ProductItem
Product
   individualPrice
```

So basically in `Invoice` I need a formula that does the standard P * Q... for each `InvoiceItem` it needs to multiply the `amount` by the `Product`'s `individualPrice`.

I've tried the following:

`$("InvoiceItems").map(I => i.Product.individualPrice)` but that comes out as empty...
If I simply return `$("InvoiceItems")`, I can see on the JSON that the nested `Product` is not populated... Am I missing something obvious here? Is this supported?

I also tried adding a formula for the item price in `InvoiceItem` to then sum those, but it did not work either.

**Comment by melohagan** [comment, marked as answer]
Hey @pbendersky 

Apologies for the late reply, I believe I have found a solution for this one!

**Product**
![Screenshot 2023-01-23 at 14 24 31](https://user-images.githubusercontent.com/101575380/214063516-0e538f5a-32bb-45c1-ab04-43da8712af9d.png)

**InvoiceItem**
![Screenshot 2023-01-23 at 14 25 57](https://user-images.githubusercontent.com/101575380/214063828-4ba3e693-4276-4684-8fef-3971a3bbab2c.png)

**Invoice**
![Screenshot 2023-01-23 at 14 26 18](https://user-images.githubusercontent.com/101575380/214063906-c9652b4e-584a-45d2-87fb-ce2dd87b310d.png)

### Step 1
Add the following **static** formula column (called 'Total') to the *InvoiceItem* table:
```js
return $("Product.0.Price") * $("Amount");
```

### Step 2
Add a formula to the *Invoice* table like so:
![Screenshot 2023-01-23 at 14 29 19](https://user-images.githubusercontent.com/101575380/214064584-8b6edda6-94a5-4c97-a045-7ba74f832e78.png)

**Note**
You will notice in the bindings helper that the 'Total' field does not appear:
![Screenshot 2023-01-23 at 14 30 39](https://user-images.githubusercontent.com/101575380/214064942-6b27927e-a36c-4cd2-9911-81ba6b50e54e.png)

This is because formula columns typically can't be accessed via relationships. **static** formulas are an exception, but are only available in the internal DB. All SQL data table formulas are **dynamic**.
